pkg/collector/corechecks/cluster/orchestrator/collector_bundle.go

```diff
@@ -192,7 +198,14 @@ func (cb *CollectorBundle) importCRDCollectorsFromCheckConfig() bool {
 	if len(cb.check.instance.CRDCollectors) == 0 {
 		return false
 	}
-	for _, c := range cb.check.instance.CRDCollectors {
+
+	crdCollectors := cb.check.instance.CRDCollectors
+	if len(cb.check.instance.CRDCollectors) > defaultMaximumCRDs {
+		crdCollectors = cb.check.instance.CRDCollectors[:defaultMaximumCRDs]
+		cb.check.Warnf("Too many crd collectors are configured, only collect first %d collectors", defaultMaximumCRDs)
```

Comment:
grammar tweak:
`will only collect the first %d collectors` or 
`only collecting the first %d collectors`